content/articles/getting-started-with-optical-character-recognition/index.md

```diff
@@ -0,0 +1,170 @@
+   [130.90935177105715, 630.1665471049322]],
+  'NEWS',
+  0.993106484413147),
+ ([[190.6717988226486, 618.007698233973],
+   [392.5179510077811, 588.6436857858664],
+   [398.3282011773514, 642.992301766027],
+   [196.48204899221892, 672.3563142141336]],
+  'COMING',
+  0.9999751310992928)]
+```
+After applying EasyOCR on the image, we can see that it has been able to extract the text from the image with a good confidence value. The different values indicate the coordinates where our text is in the image. This is okay but it'll be nice if we visualized it on the image.
+
+### Visualizing results using the OpenCV library
+Let's begin by defining a couple of key variables to determine where our different coordinates are. We'll use the OpenCV library for this task.
+
+Let's set our coordinate variable.
+
+```python
+top_left = tuple(results[0][0][0])
+bottom_right = tuple(results[0][0][2])
+text = results[0][1]
+font = cv2.FONT_HERSHEY_PLAIN
+```
+We began by defining a variable for our `top_left` coordinate. We've converted it into a `tuple` because when we pass it to OpenCV, it's expecting a tuple. We've done a similar thing with the `bottom_right` variable. We grab that text and put it into a variable known as `text`. We've also gone ahead and defined the OpenCV font that we're going to use. To learn about more OpenCV text fonts, please refer to this [article](https://www.oreilly.com/library/view/mastering-opencv-4/9781789344912/16b55e96-1027-4765-85d8-ced8fa071473.xhtml).
+
+Let's now go ahead and visualize it. 
+
+```python
+img = cv2.imread("image-one.jpg")
+img = cv2.rectangle(img,top_left,bottom_right,(0,255,0),3)
+img = cv2.putText(img,text,top_left, font, 0.5,(255,255,255),2,cv2.LINE_AA)
+plt.imshow(img)
+plt.show()
+```
+![Box](/engineering-education/getting-started-with-optical-character-recognition/box.png)
+
+Though small, we can see that a green bounding box has been drawn on the top right text. That's optical character recognition in a nutshell. Currently, it's only able to handle a single text. What happens if we have an image with multiple lines of text?
+
+Let's try and make EasyOCR handle multiple lines of the extracted text.
+
+### Handling multiple lines
+Handling it is all the same as in the previous reader code, what changes is how we go ahead and visualize it. We need to loop through to visualize the other texts.
+> As a side note, it will be taking a little longer to process as we are now working with multiple texts.
+
+```python
+image = cv2.imread("image-one.jpg")
+spacer = 150
+for detection in results: 
+    top_left_detection = tuple(detection[0][0])
+    bottom_right_detection = tuple(detection[0][2])
+    text_detection = detection[1]
+    image = cv2.rectangle(image,top_left_detection,bottom_right_detection,(255,0,0),3)
+    image = cv2.putText(image,text_detection,(20,spacer), font, 0.5,(255,0,0),2,cv2.LINE_AA)
+    spacer+=20
+    
+plt.imshow(image)
+plt.show()
+```
+Result:
```

Comment:
Running the code block for **handling multiple lines** yields the following error

```python
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
<ipython-input-22-61b6150f70c9> in <module>()
      5     bottom_right = tuple(detection[0][2])
      6     text = detection[1]
----> 7     img = cv2.rectangle(img,top_left,bottom_right,(0,255,0),3)
      8     img = cv2.putText(img,text,(20,spacer), font, 0.5,(0,255,0),2,cv2.LINE_AA)
      9     spacer+=15

error: OpenCV(4.5.4) :-1: error: (-5:Bad argument) in function 'rectangle'
> Overload resolution failed:
>  - Can't parse 'pt1'. Sequence item with index 0 has a wrong type
>  - Can't parse 'pt1'. Sequence item with index 0 has a wrong type
>  - Can't parse 'rec'. Expected sequence length 4, got 2
>  - Can't parse 'rec'. Expected sequence length 4, got 2
```